IntegrationTests/Services/AWSS3IntegrationTests/S3FlexibleChecksumsTests.swift

```diff
@@ -77,12 +77,54 @@ final class S3FlexibleChecksumsTests: S3XCTestCase {
         let data = try await body.readData()
         XCTAssertEqual(data, originalData)
     }
+    
+    // MARK: - use client config checksum
+
+    func test_putGetObject_with_client_config_checksum_crc32() async throws {
+        try await _testPutGetObject(
+            withChecksumAlgorithm: nil,
+            objectNameSuffix: "crc32-data",
+            upload: .data(originalData),
+            clientConfigChecksumAlgorithm: ChecksumAlgorithm.crc32
+        )
+    }
+    
+    func test_putGetObject_with_client_config_checksum_crc32c() async throws {
+        try await _testPutGetObject(
+            withChecksumAlgorithm: nil,
+            objectNameSuffix: "crc32c-data",
+            upload: .data(originalData),
+            clientConfigChecksumAlgorithm: ChecksumAlgorithm.crc32c
+        )
+    }
+    
+    func test_putGetObject_with_client_config_checksum_sha1() async throws {
+        try await _testPutGetObject(
+            withChecksumAlgorithm: nil,
+            objectNameSuffix: "sha1-data",
+            upload: .data(originalData),
+            clientConfigChecksumAlgorithm: ChecksumAlgorithm.sha1
```

Comment:
Same question as on smithy-swift: Should the plugin use ChecksumAlgorithm enum or the particular client's supported checksum algorithms? Ex. for S3 it is `S3ClientTypes.ChecksumAlgorithm.x` where x is crc32 etc.